core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -704,7 +705,27 @@ public int hashCode() {
         /**
          * Returns the next operation in the snapshot or <code>null</code> if we reached the end.
          */
-        Translog.Operation next() throws IOException;
+        Position next() throws IOException;
+
+    }
+
+    public static class Position {
```

Comment:
@s1monw So, I reasoned through this already. Here was my thinking, but let me know what you think? It's fine to not put the operation in the version map since it will cleared immediately after recovery and is useless during recovery. But the indexing memory controller accounting is done off the size from the translog location. Thus, the location needs to be set on the translog so the indexing memory controller listener can pick it up when accounting post an indexing operation.

I'm open to alternative solutions, but this seemed simple and clean.